### ISSUE TITLE
Parameter settings and dataset partitioning in incremental training

### ISSUE THREAD

**Issue by toni-k8s** [ultralytics/ultralytics#24601]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am facing a few challenges during the incremental training of my YOLOv26-based model and would welcome your insights.

The current state of my project is as follows: we have an already trained model, which we can refer to as the **"base model,"** alongside its corresponding **"base dataset."** My immediate goal is to introduce a **new scene** to this model. This new dataset contains only the new scene context but shares the exact same two target classes as the base model.

Regarding incremental training, I have the following questions:

1. **Data Strategy:** What is the best method to extract a subset of data from the base dataset and combine it with the new data? Is there a fast or streamlined approach to achieve this?
2. **Hyperparameters:** What specific training parameters and strategies should be configured during this incremental training phase to balance learning new scenes while preserving old knowledge?

Thank you for your assistance!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @toni-k8s, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue routed quickly, and an Ultralytics engineer will also assist you soon.

For questions about incremental training with YOLO26, we recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples, and many common training workflows may already be covered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
- your current `ultralytics` version
- your training command or script
- the model you started from
- how your base dataset and new dataset are currently partitioned
- dataset image examples
- training logs and results
- any relevant config or hyperparameter settings
- confirmation that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips)

Join the Ultralytics community where it suits you best 🤝. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by raimbekovm** [comment]
hi @toni-k8s, since the new dataset uses the same two classes as the base model, this is a domain-extension problem rather than a new-class one.

on data, merge the two datasets and train on the union, there is no built-in subset-extraction CLI so you copy images and matching `.txt` label files into one `train/` directory and point a new `data.yaml` at it (the `train:` field also accepts a list of paths if you want to keep sources separate). on the base side, sample stratified by scene rather than uniform random so each original scene type stays represented, a 2-4x base-to-new ratio is a reasonable starting point and you can increase the replay share if the scenes differ a lot. replay is the primary lever, a quick A/B on YOLO26n showed the model drifts toward the narrow scene without any base data mixed in regardless of freeze depth or lr.

on hyperparameters, start from your existing `best.pt` rather than `yolo26n.pt`, set `optimizer=AdamW, lr0=0.001` for stability, freeze the backbone with `freeze=10` (or only the detection head with `freeze=23` for very small new sets, the [fine-tuning guide](https://docs.ultralytics.com/guides/finetuning-guide/#freezing-layers) has a depth table), keep `warmup_epochs=3`, and run 10-30 epochs with `patience=10-20` since longer runs on narrow data increase forgetting. validate on a held-out split that contains both old and new scenes so degradation on either side is visible.
